proposals/collection-literals.md

```diff
@@ -112,23 +112,54 @@ Collection literals are [target-typed](https://github.com/dotnet/csharplang/blob
 
     * The behavior of collection literals with collections that are not well-behaved is undefined.
 
-## Constructible collection types (target types)
-[constructible-collection-types]: #constructible-collection-types
+## Conversions
+A *collection literal conversion* allows a collection literal expression to be converted to a type.
 
-The following are _constructible collection types_ that can be used to construct collection literals.
-These are the valid _target types_ for collection literals.
+The following implicit *collection literal conversions* exist from a collection literal expression:
 
-The list is in priority order. If a collection type belongs in multiple categories, the first is used.
+* To a single dimensional *array type* `T[]`, or a *span type* `System.Span<T>` or `System.ReadOnlySpan<T>`, where:
+  * For each *expression element* `Ei` there is an implicit conversion from `Ei` to `T`.
+  * For each *dictionary element* `Ki:Vi`, `T` is a type `System.Collections.Generic.KeyValuePair<K, V>` and there is an implicit conversion from `Ki` to `K` and from `Vi` to `V`.
+  * For each *spread element* `Si` there is an implicit conversion from the *iteration type* of `Si` to `T`.
 
-Actual translation of the literal to the corresponding type is defined [below](#collection-literal-translation).
+* To a *type* with an associated *[builder](#construct-methods)* where there is an implicit collection literal conversion from the collection literal to the *span type* of the builder argument.
 
-* Single dimensional arrays (e.g. `T[]`)
-* [*Span types*](#span-types)
-* Types with a suitable [`Construct` method](#construct-methods)
-* Types that implement `System.Collections.IDictionary` - *dictionary [collection initializers](#collection-initializers)*
-* Interface types _`I<TKey, TValue>`_ implemented by `System.Collections.Generic.Dictionary<TKey, TValue>` (e.g. `IDictionary<TKey, TValue>`, `IReadOnlyDictionary<TKey, TValue>`)
-* Types that implement `System.Collections.IEnumerable` - [*collection initializers*](#collection-initializers)
-* Interface types _`I<T>`_ implemented by `System.Collections.Generic.List<T>` (e.g. `IEnumerable<T>`, `IList<T>`, `IReadOnlyList<T>`)
+* To a *type* that implements `System.Collections.IDictionary` where:
+  * The *type* contains an applicable instance constructor that can be invoked with no arguments or invoked with a single argument for the 0-th parameter where the parameter has type `System.Int32` and name `capacity`.
+  * For each *expression element* `Ei`:
+    * the type of `Ei` is `dynamic` and there is an applicable indexer setter that can be invoked with two `dynamic` arguments, or
```

Comment:
I'm not sure the last part can actually be required? We don't check to see if an indexer exists with `dynamic`, we let the runtime binder figure that out.

Reply:
I believe the compiler can still check for a property group: see [sharplab.io](EYLgZgpghgLgrgJwgZwLQEkB2AbAlpiAQQQSgE9kAaAExAGoAfAAQCYBGAWAChuA3KBAAJqggLyCCAd0EBZMgBFcAYxi4A9pgFkAFAEoA3N2plNAW2WCIYiRGlrgAKwgq9hnlwD616gG0IAXTcvcV8ISkEABkDublZZBWVVDS1uAG9uQUzBJgBmQXsnFUEYAAtcZB98GEEAD3CqwTJ).

Reply:
I believe the compiler can still check for a property group: see [sharplab.io](EYLgZgpghgLgrgJwgZwLQEkB2AbAlpiAQQQSgE9kAaAExAGoAfAAQCYBGAWAChuA3KBAAJqggLyCCAd0EBZMgBFcAYxi4A9pgFkAFAEoA3N2plNAW2WCIYiRGlrgAKwgq9hnlwD616gG0IAXTcvcV8ISkEABkDublZZBWVVDS1uAG9uQUzBJgBmQXsnFUEYAAtcZB98GEEAD3CqwTJ).

Reply:
Huh, TIL. It might be good to reference that applicability is defined by [12.6.5](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/expressions.md#1265-compile-time-checking-of-dynamic-member-invocation) of the spec, so it's not ambiguous whether you mean that the function must itself take a `dynamic` argument.